dakara_server/library/views.py

```diff
@@ -276,7 +276,7 @@ def get_queryset(self):
             # saving the parsed query to give it back to the client
             self.query_parsed = {'remaining': res}
 
-        return query_set.order_by(Lower("title"), Lower("subtitle"))
+        return query_set.distinct().order_by(Lower("title"), Lower("subtitle"))
```

Comment:
There is something odd with this `distinct`. If I remove it, the test `library.tests_work.WorkListViewAPIViewTestCase.test_get_work_list_with_query` fails (an only this one), because the searched work is given twice. It seems that line 267 is responsible for that, even though `Q(alternative_title__title__icontains=remain)` should not match anything, since for this test `remain` corresponds to the work title.

Reply:
Anyway, `distinct` prevents this from happening, so this is fine. I am a bit puzzled by the filtering, though.

Reply:
I don't think it is necessary odd if we consider that Django makes a SQL joint between Work and WorkAlternativeTitle.

Reply:
I don't get it.